File: app/src/main/java/com/proxichat/app/ui/settings/SettingsScreen.kt
package com.proxichat.app.ui.settings

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.ListItem
import androidx.compose.material3.ListItemDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Switch
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.material3.TopAppBar
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.hilt.navigation.compose.hiltViewModel

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun SettingsScreen(
    onNavigateBack: () -> Unit,
    viewModel: SettingsViewModel = hiltViewModel()
) {
    val state by viewModel.uiState.collectAsState()
    var showNameDialog by remember { mutableStateOf(false) }
    var showDarkModeDialog by remember { mutableStateOf(false) }
    var showClearHistoryDialog by remember { mutableStateOf(false) }
    var editedName by remember(state.displayName) { mutableStateOf(state.displayName) }

    Scaffold(
        topBar = {
            TopAppBar(
                title = {
                    Text(
                        text = "Settings",
                        fontWeight = FontWeight.Bold
                    )
                },
                navigationIcon = {
                    IconButton(onClick = onNavigateBack) {
                        Icon(
                            imageVector = Icons.AutoMirrored.Filled.ArrowBack,
                            contentDescription = "Back"
                        )
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MaterialTheme.colorScheme.surface
                )
            )
        }
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues)
                .verticalScroll(rememberScrollState())
        ) {
            // Profile section
            SectionTitle("Profile")

            ListItem(
                headlineContent = { Text("Display Name") },
                supportingContent = { Text(state.displayName) },
                modifier = Modifier.clickable { showNameDialog = true },
                colors = ListItemDefaults.colors(containerColor = Color.Transparent)
            )

            HorizontalDivider(modifier = Modifier.padding(horizontal = 16.dp))

            // Appearance section
            SectionTitle("Appearance")

            ListItem(
                headlineContent = { Text("Dark Mode") },
                supportingContent = {
                    Text(
                        when (state.darkMode) {
                            "on" -> "Always on"
                            "off" -> "Always off"
                            else -> "Follow system"
                        }
                    )
                },
                modifier = Modifier.clickable { showDarkModeDialog = true },
                colors = ListItemDefaults.colors(containerColor = Color.Transparent)
            )

            HorizontalDivider(modifier = Modifier.padding(horizontal = 16.dp))

            // Bluetooth section
            SectionTitle("Bluetooth")

            ListItem(
                headlineContent = { Text("Discoverable") },
                supportingContent = { Text("Allow nearby devices to find you") },
                trailingContent = {
                    Switch(
                        checked = state.isDiscoverable,
                        onCheckedChange = viewModel::setDiscoverable
                    )
                },
                colors = ListItemDefaults.colors(containerColor = Color.Transparent)
            )

            ListItem(
                headlineContent = { Text("Auto-reconnect") },
                supportingContent = { Text("Automatically reconnect to known devices") },
                trailingContent = {
                    Switch(
                        checked = state.autoReconnect,
                        onCheckedChange = viewModel::setAutoReconnect
                    )
                },
                colors = ListItemDefaults.colors(containerColor = Color.Transparent)
            )

            HorizontalDivider(modifier = Modifier.padding(horizontal = 16.dp))

            // Security section
            SectionTitle("Security")

            ListItem(
                headlineContent = { Text("Message Encryption") },
                supportingContent = { Text("Encrypt messages with AES-256-GCM") },
                trailingContent = {
                    Switch(
                        checked = state.encryptionEnabled,
                        onCheckedChange = viewModel::setEncryption
                    )
                },
                colors = ListItemDefaults.colors(containerColor = Color.Transparent)
            )

            HorizontalDivider(modifier = Modifier.padding(horizontal = 16.dp))

            // Data section
            SectionTitle("Data")

            ListItem(
                headlineContent = {
                    Text(
                        "Clear Chat History",
                        color = MaterialTheme.colorScheme.error
                    )
                },
                supportingContent = { Text("Delete all messages") },
                modifier = Modifier.clickable { showClearHistoryDialog = true },
                colors = ListItemDefaults.colors(containerColor = Color.Transparent)
            )

            HorizontalDivider(modifier = Modifier.padding(horizontal = 16.dp))

            // About section
            SectionTitle("About")

            ListItem(
                headlineContent = { Text("Version") },
                supportingContent = { Text(state.appVersion) },
                colors = ListItemDefaults.colors(containerColor = Color.Transparent)
            )

            ListItem(
                headlineContent = { Text("ProxiChat") },
                supportingContent = { Text("Bluetooth proximity chat — no internet required") },
                colors = ListItemDefaults.colors(containerColor = Color.Transparent)
            )

            Spacer(modifier = Modifier.height(32.dp))
        }
    }

    // Edit name dialog
    if (showNameDialog) {
        AlertDialog(
            onDismissRequest = { showNameDialog = false },
            title = { Text("Display Name") },
            text = {
                OutlinedTextField(
                    value = editedName,
                    onValueChange = { if (it.length <= 20) editedName = it },
                    singleLine = true,
                    modifier = Modifier.fillMaxWidth(),
                    shape = MaterialTheme.shapes.medium,
                    supportingText = { Text("${editedName.length}/20") }
                )
            },
            confirmButton = {
                TextButton(
                    onClick = {
                        if (editedName.isNotBlank()) {
                            viewModel.updateDisplayName(editedName.trim())
                        }
                        showNameDialog = false
                    }
                ) { Text("Save") }
            },
            dismissButton = {
                TextButton(onClick = { showNameDialog = false }) { Text("Cancel") }
            }
        )
    }

    // Dark mode dialog
    if (showDarkModeDialog) {
        AlertDialog(
            onDismissRequest = { showDarkModeDialog = false },
            title = { Text("Dark Mode") },
            text = {
                Column {
                    listOf(
                        "system" to "Follow system",
                        "on" to "Always on",
                        "off" to "Always off"
                    ).forEach { (value, label) ->
                        ListItem(
                            headlineContent = { Text(label) },
                            modifier = Modifier.clickable {
                                viewModel.setDarkMode(value)
                                showDarkModeDialog = false
                            },
                            colors = ListItemDefaults.colors(
                                containerColor = if (state.darkMode == value)
                                    MaterialTheme.colorScheme.primaryContainer.copy(alpha = 0.3f)
                                else Color.Transparent
                            )
                        )
                    }
                }
            },
            confirmButton = {
                TextButton(onClick = { showDarkModeDialog = false }) { Text("Cancel") }
            }
        )
    }

    // Clear history confirmation dialog
    if (showClearHistoryDialog) {
        AlertDialog(
            onDismissRequest = { showClearHistoryDialog = false },
            title = { Text("Clear Chat History") },
            text = { Text("This will permanently delete all messages. This action cannot be undone.") },
            confirmButton = {
                TextButton(
                    onClick = {
                        viewModel.clearChatHistory()
                        showClearHistoryDialog = false
                    }
                ) {
                    Text(
                        "Delete All",
                        color = MaterialTheme.colorScheme.error
                    )
                }
            },
            dismissButton = {
                TextButton(onClick = { showClearHistoryDialog = false }) { Text("Cancel") }
            }
        )
    }
}

@Composable
private fun SectionTitle(title: String) {
    Text(
        text = title,
        style = MaterialTheme.typography.titleSmall,
        fontWeight = FontWeight.SemiBold,
        color = MaterialTheme.colorScheme.primary,
        modifier = Modifier.padding(start = 16.dp, top = 24.dp, bottom = 8.dp)
    )
}

File: flutter_app/lib/screens/settings_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../bluetooth/ble_controller.dart';
import '../data/message_store.dart';
import '../data/preferences.dart';

class SettingsScreen extends StatelessWidget {
  const SettingsScreen({super.key});

  @override
  Widget build(BuildContext context) {
    final prefs = context.watch<UserPreferences>();
    final ble = context.read<BleController>();
    final store = context.read<MessageStore>();

    return Scaffold(
      appBar: AppBar(title: const Text('Settings', style: TextStyle(fontWeight: FontWeight.bold))),
      body: ListView(children: [
        _Section('Profile'),
        ListTile(
          title: const Text('Display Name'),
          subtitle: Text(prefs.displayName),
          onTap: () => _editName(context, prefs, ble),
        ),

        const Divider(indent: 16, endIndent: 16),
        _Section('Appearance'),
        ListTile(
          title: const Text('Dark Mode'),
          subtitle: Text(switch (prefs.darkMode) { 'on' => 'Always on', 'off' => 'Always off', _ => 'Follow system' }),
          onTap: () => _pickDarkMode(context, prefs),
        ),

        const Divider(indent: 16, endIndent: 16),
        _Section('Bluetooth'),
        SwitchListTile(
          title: const Text('Discoverable'),
          subtitle: const Text('Allow nearby devices to find you'),
          value: prefs.discoverable,
          onChanged: (v) {
            prefs.setDiscoverable(v);
            if (v) ble.startAdvertising(); else ble.stopAdvertising();
          },
        ),

        const Divider(indent: 16, endIndent: 16),
        _Section('Data'),
        ListTile(
          title: Text('Clear Chat History', style: TextStyle(color: Theme.of(context).colorScheme.error)),
          onTap: () => showDialog(
            context: context,
            builder: (_) => AlertDialog(
              title: const Text('Clear Chat History'),
              content: const Text('Delete all messages? This cannot be undone.'),
              actions: [
                TextButton(onPressed: () => Navigator.pop(context), child: const Text('Cancel')),
                TextButton(
                  onPressed: () { store.deleteAll(); Navigator.pop(context); },
                  child: Text('Delete', style: TextStyle(color: Theme.of(context).colorScheme.error)),
                ),
              ],
            ),
          ),
        ),

        const Divider(indent: 16, endIndent: 16),
        _Section('About'),
        const ListTile(title: Text('Version'), subtitle: Text('1.0.0 (Flutter)')),
      ]),
    );
  }

  void _editName(BuildContext context, UserPreferences prefs, BleController ble) {
    final ctrl = TextEditingController(text: prefs.displayName);
    showDialog(
      context: context,
      builder: (_) => AlertDialog(
        title: const Text('Display Name'),
        content: TextField(controller: ctrl, maxLength: 20, autofocus: true),
        actions: [
          TextButton(onPressed: () => Navigator.pop(context), child: const Text('Cancel')),
          TextButton(
            onPressed: () {
              final name = ctrl.text.trim();
              if (name.isNotEmpty) {
                prefs.setDisplayName(name);
                ble.updateDisplayName(name);
              }
              Navigator.pop(context);
            },
            child: const Text('Save'),
          ),
        ],
      ),
    );
  }

  void _pickDarkMode(BuildContext context, UserPreferences prefs) {
    showDialog(
      context: context,
      builder: (_) => SimpleDialog(
        title: const Text('Dark Mode'),
        children: [
          for (final entry in {'system': 'Follow System', 'on': 'Always On', 'off': 'Always Off'}.entries)
            RadioListTile<String>(
              title: Text(entry.value),
              value: entry.key,
              groupValue: prefs.darkMode,
              onChanged: (v) { prefs.setDarkMode(v!); Navigator.pop(context); },
            ),
        ],
      ),
    );
  }
}

class _Section extends StatelessWidget {
  final String title;
  const _Section(this.title);
  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.fromLTRB(16, 24, 16, 8),
      child: Text(title, style: Theme.of(context).textTheme.titleSmall?.copyWith(fontWeight: FontWeight.w600, color: Theme.of(context).colorScheme.primary)),
    );
  }
}
